src/main.zig

```diff
@@ -353,6 +353,64 @@ fn add_cert_public_key(state: *VerifierCaptureState, _: u8, length: usize, reade
     };
 }
 
+fn add_cert_extensions(state: *VerifierCaptureState, tag: u8, length: usize, reader: anytype) !void {
+    const schema = .{
+        .sequence_of,
+        .{ .capture, 0, .sequence },
+    };
+    const captures = .{
+        state, add_cert_extension,
+    };
+
+    try asn1.der.parse_schema(schema, captures, reader);
+}
+
+fn add_cert_extension(state: *VerifierCaptureState, tag: u8, length: usize, reader: anytype) !void {
+    const start = state.fbs.pos;
+
+    // The happy path is allocation free
+    // TODO: add a preflight check to mandate a specific tag
+    const object_id = try asn1.der.parse_value(state.allocator, reader);
+    defer object_id.deinit(state.allocator);
+    if (object_id != .object_identifier) return error.DoesNotMatchSchema;
+    if (object_id.object_identifier.len != 4)
+        return;
+
+    const data = object_id.object_identifier.data;
+    // Prefix == id-ce
+    if (data[0] != 2 or data[1] != 5 or data[2] != 29)
```

Comment:
These magic numbers are copied from x509. We should probably define them somewhere >_>

Reply:
Yes, I will go over the source code once I add client certificates and pull a bunch of the object identifiers we check against out into a new file.